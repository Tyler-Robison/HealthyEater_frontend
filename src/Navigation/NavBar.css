.NavBar {
    background-color: rgb(252, 184, 58);
    color: blue;
    /* border: black solid 2px; */
    /* display: flex;
    flex-direction: row; */
    border-bottom: 2px blue solid;
    /* position: relative; */
}

.nav-link.active {
    font-weight: 700;
    text-decoration: underline;
}

.NavBar-Brand {
    margin-left: 3rem;
    padding-right: 8rem;
    color: aqua;
}

/* .logout-btn {
    position: absolute;
    right: 3rem;
} */

.login {
    font-size: 1.2rem;
}

.signup {
    font-size: 1.2rem;
}

.brand-img {
    margin-left: 0.2rem;
}

/* .test {
    display: flex; 
    flex-direction: row;
    justify-content:flex-end;
} */

.NavBar-Container {
    display: flex; 
    flex-direction: row;
    justify-content:flex-end;
}