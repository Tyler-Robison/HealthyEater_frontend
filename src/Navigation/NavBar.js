import React, { useContext } from "react";
import { NavLink } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.min.css"
import GlobalContext from "../context/GlobalContext";
import { Navbar, Container } from "react-bootstrap";
import lettuce from '../Images/lettuce.png'
import './NavBar.css'

/** NavBar is always visible, allows quick navigation between the site's main features
 * 
 * loggedOutView - dispays login/sign-up and continue as guest button
 * 
 * loggedInView - displays the four main features of the site (Find Recipes, Saved Recipes, Mealplan, Calculate Points) and logout button
  */
const NavBar = ({ logout, guestLogin }) => {
    const { currentUser } = useContext(GlobalContext)

    const loggedInView = <>

        <NavLink className="nav-link" to="/findrecipes">
            Find Recipes
        </NavLink>

        <NavLink className="nav-link" to="/savedrecipes">
            Saved Recipes
        </NavLink>

        <NavLink className="nav-link" to="/mealplan">
            Mealplan
        </NavLink>

        <NavLink className="nav-link" to="/points">
            Calculate Points
        </NavLink>


        {currentUser && <button className="general-btn-red logout-btn" onClick={logout}> Log Out {currentUser.username}</button>}

    </>

    const loggedOutView = <>

   

            <NavLink className="nav-link signup " to="/signup">
                Signup
            </NavLink>

            <button onClick={guestLogin} className="general-btn mt-2">Continue as Guest</button>

            <NavLink className="nav-link login" to="/login">
                Login
            </NavLink>
   

    </>

    return (
        <Navbar className="NavBar" expand="lg">

            <Navbar.Brand className="NavBar-Brand">
                <NavLink className="nav-link" to="/healthyeater">
                    Healthy-Eater<img className="brand-img" src={lettuce} height='28px' width='28px'></img>
                </NavLink>
            </Navbar.Brand>

            <Container className="NavBar-Container">

                {currentUser ? loggedInView : loggedOutView}

            </Container>
        </Navbar>
    )
}

export default NavBar;