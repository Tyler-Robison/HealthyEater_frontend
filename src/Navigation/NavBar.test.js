import { render, screen } from '@testing-library/react';
import NavBar from './NavBar';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../testContext'


it("renders without crashing", function () {
    <MemoryRouter>
        <NavBar />
    </MemoryRouter>
});

it("matches snapshot when logged in", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider>
                <NavBar />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider userData={null} >
                <NavBar />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});