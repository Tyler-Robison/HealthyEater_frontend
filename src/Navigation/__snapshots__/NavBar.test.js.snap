// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <nav
    class="NavBar navbar navbar-expand-lg navbar-light"
  >
    <span
      class="NavBar-Brand navbar-brand"
    >
      <a
        class="nav-link"
        href="/healthyeater"
      >
        Healthy-Eater
        <img
          class="brand-img"
          height="28px"
          src="lettuce.png"
          width="28px"
        />
      </a>
    </span>
    <div
      class="NavBar-Container container"
    >
      <a
        class="nav-link"
        href="/findrecipes"
      >
        Find Recipes
      </a>
      <a
        class="nav-link"
        href="/savedrecipes"
      >
        Saved Recipes
      </a>
      <a
        class="nav-link"
        href="/mealplan"
      >
        Mealplan
      </a>
      <a
        class="nav-link"
        href="/points"
      >
        Calculate Points
      </a>
      <button
        class="general-btn-red logout-btn"
      >
         Log Out Tyler
      </button>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    class="NavBar navbar navbar-expand-lg navbar-light"
  >
    <span
      class="NavBar-Brand navbar-brand"
    >
      <a
        class="nav-link"
        href="/healthyeater"
      >
        Healthy-Eater
        <img
          class="brand-img"
          height="28px"
          src="lettuce.png"
          width="28px"
        />
      </a>
    </span>
    <div
      class="NavBar-Container container"
    >
      <a
        class="nav-link signup "
        href="/signup"
      >
        Signup
      </a>
      <button
        class="general-btn"
      >
        Continue as Guest
      </button>
      <a
        class="nav-link login"
        href="/login"
      >
        Login
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
